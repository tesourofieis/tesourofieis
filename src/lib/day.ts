import { getDate, getDay, getMonth } from "date-fns";
import {
  COMMUNE_C_10A,
  COMMUNE_C_10B,
  COMMUNE_C_10C,
  COMMUNE_C_10PASC,
  COMMUNE_C_10T,
  PATTERN_ADVENT,
  PATTERN_EASTER,
  PATTERN_EPIPHANY,
  PATTERN_LENT,
  PATTERN_POST_EPIPHANY_SUNDAY,
  PATTERN_PRE_LENTEN,
  TABLE_OF_PRECEDENCE,
  TEMPORA_RANK_MAP,
  TITLES,
  TYPE_TEMPORA,
} from "./constants.ts";

// """ Class used to keep `Observance` objects for particular days of Missal.
//
// It contains three lists: `tempora`, `celebration` and `commemoration`.
// On Missal's creation the lists are filled in so that `tempora` always contains `Observance` representing
// given variable day, `celebration` contains an `Observance`s to be celebrated in this day and
// `commemoration` contains zero or more `Observance`s that should be commemorated with the main celebration.
// """
class Day {
  date: string;
  tempora: Observance[] = [];
  celebration: Observance[] = [];
  commemoration: Observance[] = [];

  constructor(date: string) {
    this.date = date;
  }

  get all(): Observance[] {
    return this.tempora.concat(this.celebration).concat(this.commemoration);
  }

  getTemporaName() {
    return this.tempora[0]?.title;
  }

  serialize() {
    const serialized = {};
    const containers = ["tempora", "celebration", "commemoration"];

    for (const container of containers) {
      serialized[container] = this[container].map((item: Day) =>
        item.serialize(),
      );
    }

    return serialized;
  }
}

class Observance {
  date: string;
  flexibility: string;
  name: string;
  rank: number;
  colors: string[];
  id: string;
  link: string;
  title: string;
  priority: number | null;
  weekday: number | null;

  // """ Build an Observance out of identifier and calendar date
  //
  // :param observance_id: observance's identifier in format
  //                <flexibility>:<identifier>:<rank>
  // :type observance_id: string
  // :param date_: specific date in which the observance is supposed
  //             to be placed. For some Sancti days its rank (class)
  //             depends on which calendar day they occur.
  // :type date_: `date ` object
  // """
  constructor(observanceId: string, date_: string) {
    const [flexibility, name, rank, color] = observanceId.split(":");
    this.date = date_;
    this.flexibility = flexibility;
    this.name = name;
    this.rank = this.calcRank(observanceId, Number.parseInt(rank, 10));
    this.colors = Array.from(color);
    this.id = `${this.flexibility}:${this.name}:${this.rank}:${color}`;
    this.link = `missal/${this.getTempora(this.name)}/${this.name}`;
    this.title = TITLES[observanceId];

    // Determine the weekday attribute based on the type of observance
    if (
      flexibility === TYPE_TEMPORA &&
      ![
        COMMUNE_C_10A,
        COMMUNE_C_10B,
        COMMUNE_C_10C,
        COMMUNE_C_10PASC,
        COMMUNE_C_10T,
      ].includes(observanceId)
    ) {
      if (name.replace(/^.*-(\d+).*$/, "$1") === "10-Dur") {
        this.weekday = 0;
      }
      this.weekday = Number.parseInt(name.replace(/^.*-(\d+).*$/, "$1"), 10);
    } else {
      this.weekday = getDay(this.date);
    }

    this.priority = this.calcPriority();
  }

  private getTempora(name: string): string {
    if (this.flexibility === "tempora") {
      if (name.includes("nat")) {
        return "natal";
      }
      if (name.includes("adv")) {
        return "advento";
      }
      if (name.includes("pasc")) {
        return "pascoa";
      }
      if (name.includes("epi")) {
        return "epifania";
      }
      if (name.includes("quadp")) {
        return "pre-quaresma";
      }
      if (name.includes("quad")) {
        return "quaresma";
      }
      if (name.includes("pent")) {
        return "pentecostes";
      }
    }

    if (this.flexibility === "santos") {
      console.log(`Matched Santos: ${name}`);
      return "santos";
    }

    if (this.flexibility === "commune") {
      return "comum";
    }

    if (this.flexibility === "votivas") {
      return "votivas";
    }

    console.log(`Checking name: ${name}, flexibility: ${this.flexibility}`);
    throw Error(`No directory for ${name}, flexibility , ${this.flexibility}`);
  }

  private calcRank(observanceId: string, originalRank: number): number {
    for (const caseItem of TEMPORA_RANK_MAP) {
      if (
        getMonth(this.date) === caseItem.month &&
        getDate(this.date) === caseItem.day &&
        new RegExp(caseItem.pattern).test(observanceId)
      ) {
        return caseItem.rank;
      }
    }
    return originalRank;
  }

  private calcPriority() {
    for (let priority = 0; priority < TABLE_OF_PRECEDENCE.length; priority++) {
      if (this.id.match(TABLE_OF_PRECEDENCE[priority])) {
        return priority;
      }
    }
  }

  serialize() {
    return {
      id: this.id,
      link: this.link,
      rank: this.rank,
      title: this.title,
      colors: this.colors,
    };
  }
}

export { Day, Observance };
