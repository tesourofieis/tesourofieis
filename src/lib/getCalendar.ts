import { Calendar } from "./src/calendar.ts";

function getCalendar(year: number) {
  const calendar = new Calendar(year);

  return calendar;
}

function getCalendarDay(year: number, date: string) {
  const calendar = new Calendar(year).get(date);

  return calendar;
}

export { getCalendar, getCalendarDay };
