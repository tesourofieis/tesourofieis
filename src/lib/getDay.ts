import { getYear } from "date-fns";
import { Calendar } from "./calendar.ts";
import { format_propers } from "./utils.ts";

function getDay(date: string) {
  const calendar = new Calendar(getYear(date));

  const day = calendar.get(date);

  const proper = day?.getProper(calendar);

  const format = format_propers(proper);

  return format;
}

export { getDay };
