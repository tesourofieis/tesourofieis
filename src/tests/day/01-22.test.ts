import { expect, test } from "vitest";
import { getDay } from "../../lib/getDay";

test("2024-01-22", () => {
  const proper = getDay("2024-01-22");

  expect(proper).toMatchSnapshot();
});

test("2024-02-25", () => {
  const proper = getDay("2024-02-25");

  expect(proper).toMatchSnapshot();
});
