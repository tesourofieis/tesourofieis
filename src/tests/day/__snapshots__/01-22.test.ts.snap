// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`2024-01-22 1`] = `
[
  {
    "info": {
      "colors": [
        "r",
      ],
      "date": "2024-01-22",
      "day": Day {
        "celebration": [
          Observance {
            "colors": [
              "r",
            ],
            "date": "2024-01-22",
            "flexibility": "sancti",
            "id": "sancti:01-22:3:r",
            "name": "01-22",
            "priority": 79,
            "rank": 3,
            "title": "S.S. Vicente e Anastácio, Mártires",
            "weekday": 1,
          },
        ],
        "commemoration": [],
        "date": "2024-01-22",
        "tempora": [
          Observance {
            "colors": [
              "g",
            ],
            "date": "2024-01-22",
            "flexibility": "tempora",
            "id": "tempora:Epi3-1:4:g",
            "name": "Epi3-1",
            "priority": 80,
            "rank": 4,
            "title": "Segunda-feira da 3ª semana após a Epifania",
            "weekday": 1,
          },
        ],
      },
      "id": "sancti:01-22:3:r",
      "rank": 3,
      "tempora": "Segunda-feira da 3ª semana após a Epifania",
      "title": "S.S. Vicente e Anastácio, Mártires",
    },
    "sections": [
      {
        "body": {
          "latin": [
            "Ps 78:11-12; 78:10",
            "Intret in conspéctu tuo, Dómine, gémitus compeditórum: redde vicínis nostris séptuplum in sinu eórum: víndica sánguinem Sanctórum tuórum, qui effúsus est.",
            "Ps 78:1",
            "Deus, venérunt gentes in hereditátem tuam: polluérunt templum sanctum tuum: posuérunt Jerúsalem in pomórum custódiam.",
            "Glória Patri…",
            "Intret in conspéctu tuo, Dómine, gémitus compeditórum: redde vicínis nostris séptuplum in sinu eórum: víndica sánguinem Sanctórum tuórum, qui effúsus est.",
            "",
          ],
          "vernacular": [
            "Sl 78:11-12; 78:10",
            "Que os gemidos dos cativos cheguem à vossa presença, Senhor. Castigai os nossos inimigos sete vezes por cada injúria que nos têm feito: vingai o sangue que os vossos Santos derramaram.",
            "Sl 78:1",
            "Ó Deus, os povos invadiram a vossa herança, profanaram o vosso sagrado templo e reduziram Jerusalém a um monte de ruínas!",
            "Glória ao Pai.",
            "Que os gemidos dos cativos cheguem à vossa presença, Senhor. Castigai os nossos inimigos sete vezes por cada injúria que nos têm feito: vingai o sangue que os vossos Santos derramaram.",
            "",
            "",
          ],
        },
        "id": "Introitus",
      },
      {
        "body": {
          "latin": [
            "Adésto, Dómine, supplicatiónibus nostris: ut, qui ex iniquitáte nostra reos nos esse cognóscimus, beatórum Mártyrum tuórum Vincéntii et Anastásii intercessióne liberémur.",
            "Per Dominum…",
            "",
          ],
          "vernacular": [
            "Senhor, dignai-Vos ouvir as nossas súplicas, a fim de que, reconhecendo-nos réus diante de Vós pelas nossas iniquidades, sejamos livres delas por intercessão dos vossos B. B. Mártires Vicente e Anastácio.",
            "Por nosso Senhor...",
            "",
          ],
        },
        "id": "Oratio",
      },
      {
        "body": {
          "latin": [
            "Léctio libri Sapiéntiæ",
            "Sap 3:1-8",
            "Justorum ánimæ in manu Dei sunt, et non tanget illos torméntum mortis. Visi sunt oculis insipiéntium mori: et æstimála est afflíctio exitus illórum: et quod a nobis est iter, extermínium: illi autem sunt in pace. Et si coram homínibus torménta passi sunt, spes illórum immortalitáte plena est. In paucis vexáti, in multis bene disponéntur: quóniam Deus tentávit eos, et invenit illos dignos se. Tamquam aurum in fornáce probávit illos, et quasi holocáusti hóstiam accépit illos, et in témpore erit respéctus illorum. Fulgébunt justi, et tamquam scintíllæ in arundinéto discúrrent. Judicábunt natiónes, et dominabúntur pópulis, et regnábit Dóminus illórum in perpétuum.",
            "",
          ],
          "vernacular": [
            "Lição do Livro da Sabedoria.",
            "Sb 3:1-8",
            "As almas dos justos estão nas mãos de Deus; por isso o tormento da morte os não tocará. Pareciam mortos aos olhos dos insensatos: a sua saída do mundo parecia uma aflição; a sua separação de nós uma calamidade; mas, agora, estão em paz; e, ainda que tenham sofrido diante dos homens, a sua esperança está toda na imortalidade. Depois de haverem sofrido uma pena ligeira, receberam uma grande recompensa, pois Deus provou-os e achou-os dignos de si. Provou-os, como ao ouro, na fornalha; recebeu-os, como uma hóstia de holocausto; e para eles olhará benigno, quando vier o seu tempo. Os justos brilharão e resplandecerão, como as chamas, que se ateiam entre os canaviais. Eles julgarão as nações e dominarão os povos; e o Senhor reinará com eles para sempre.",
            "",
            "",
          ],
        },
        "id": "Lectio",
      },
      {
        "body": {
          "latin": [
            "Exod 15:11",
            "Gloriósus Deus in Sanctis suis: mirábilis in majestáte, fáciens prodígia.",
            "Exod 15:6",
            "℣. Déxtera tua, Dómine, glorificáta est in virtúte: déxtera manus tua confrégit inimícos. Allelúja, allelúja.",
            "Eccl 44:14",
            "Córpora Sanctórum in pace sepúlta sunt, et nómina eórum vivent in generatiónem et generatiónem. Allelúja.",
            "",
          ],
          "vernacular": [
            "Ex 15:11",
            "Deus é glorioso em seus Santos: e admirável na sua majestade, praticando prodígios.",
            "Ex 15:6",
            "℣. Senhor, a vossa dextra engrandeceu-se pela sua força: a vossa dextra esmagou os inimigos. Aleluia, aleluia.",
            "Sb 44:14",
            "Senhor, os corpos dos vossos Santos foram sepultados em paz e o seu nome subsistirá de geração em geração. Aleluia.",
            "",
            "",
          ],
        },
        "id": "Graduale",
      },
      {
        "body": {
          "latin": [
            "Exod 15:11",
            "Gloriósus Deus in Sanctis suis: mirábilis in majestáte, fáciens prodígia.",
            "Exod 15:6",
            "℣. Déxtera tua, Dómine, glorificáta est in virtúte: déxtera manus tua confrégit inimícos.",
            "Ps 125:5-6.",
            "Qui séminant in lácrimis, in gáudio metent.",
            "℣. Eúntes ibant et flébant, mitténtes sémina sua.",
            "℣. Veniéntes autem vénient cum exsultatióne, portántes manípulos suos.",
            "",
          ],
          "vernacular": [
            "Ex 15:11",
            "Deus é glorioso em seus Santos: e admirável na sua majestade, praticando prodígios.",
            "Ex 15:6",
            "℣. Senhor, a vossa dextra engrandeceu-se pela sua força: a vossa dextra esmagou os inimigos.",
            "Sl 125:5-6.",
            "Aqueles que semeiam com lágrimas ceifarão com júbilo.",
            "℣. Iam, caminhavam e lançavam a semente à terra, chorando.",
            "℣. Porém, quando voltavam, exultavam de alegria, trazendo os seus molhos de trigo.",
            "",
          ],
        },
        "id": "Tractus",
      },
      {
        "body": {
          "latin": [
            "Sequéntia ☩ sancti Evangélii secúndum Lucam.",
            "Luc 21:9-19",
            "In illo témpore: Dixit Jesus discípulis suis: Cum audieritis proelia et seditiónes, nolíte terréri: opórtet primum hæc fíeri, sed nondum statim finis. Tunc dicébat illis: Surget gens contra gentem, et regnum advérsus regnum. Et terræmótus magni erunt per loca, et pestiléntiæ, et fames, terrorésque de coelo, et signa magna erunt. Sed ante hæc ómnia injícient vobis manus suas, et persequéntur tradéntes in synagógas et custódias, trahéntes ad reges et praesides propter nomen meum: contínget autem vobis in testimónium. Pónite ergo in córdibus vestris non præmeditári, quemádmodum respondeátis. Ego enim dabo vobis os et sapiéntiam, cui non potérunt resístere et contradícere omnes adversárii vestri. Tradémini autem a paréntibus, et frátribus, et cognátis, et amícis, et morte affícient ex vobis: et éritis ódio ómnibus propter nomen meum: et capíllus de cápite vestro non períbit. In patiéntia vestra possidébitis ánimas vestras.",
            "",
          ],
          "vernacular": [
            "Continuação ☩ do santo Evangelho segundo S. Lucas.",
            "Lc 21:9-19",
            "Naquele tempo, disse Jesus aos discípulos: «Quando ouvirdes falar em guerras e sedições, não vos assusteis; pois é necessário que estas cousas aconteçam, primeiramente; mas isto não será logo o fim». «Então, dizia-lhes Ele, levantar-se-á povo contra povo e reino contra reino; em diversos lugares haverá tremores de terra, peste, fome e também aparecerão cousas espantosas, grandes sinais no céu e outros prodígios. Mas, antes que tudo isto aconteça, lançar-vos-ão as mãos e perseguir-vos-ão, entregando-vos às sinagogas, lançando-vos nas prisões e conduzindo-vos à força diante dos reis e dos governadores, por causa do meu nome. Isto acontecerá para que deis testemunho da verdade. Gravai, pois, no vosso coração este pensamento: não premediteis de que modo haveis de responder, porque vos darei palavras e sabedoria, a que os vossos inimigos não poderão resistir, nem responder. Sereis entregues pelos vossos próprios pais, irmãos, parentes e amigos, que darão a morte a alguns de vós. Sereis aborrecidos de todos, por causa do meu nome; todavia, não se perderá nem um só cabelo das vossas cabeças. Com a vossa paciência possuireis as vossas almas».",
            "",
            "",
          ],
        },
        "id": "Evangelium",
      },
      {
        "body": {
          "latin": [
            "Ps 67:36",
            "Mirábilis Deus in Sanctis suis: Deus Israël, ipse dabit virtútem et fortitúdinem plebi suæ: benedíctus Deus, allelúja.",
            "(sed post Septuagesimam dicitur)",
            "Deus.",
            "",
          ],
          "vernacular": [
            "Sl 67:36",
            "Deus é admirável em seus Santos. Deus de Israel dará ao seu povo a força e a coragem. Bendito Ele seja, pois. Aleluia.",
            "",
            "",
          ],
        },
        "id": "Offertorium",
      },
      {
        "body": {
          "latin": [
            "Múnera tibi, Dómine, nostræ devotiónis offérimus: quæ et pro tuórum tibi grata sint honóre Justórum, et nobis salutária, te miseránte, reddántur.",
            "Per Dominum…",
            "",
          ],
          "vernacular": [
            "Senhor, Vos oferecemos estes dons da nossa devoção; e que em consideração dos vossos Santos eles Vos sejam agradáveis, e pela vossa misericórdia nos sejam salutares.",
            "Por nosso Senhor...",
            "",
          ],
        },
        "id": "Secreta",
      },
      {
        "body": {
          "latin": [
            "Communis",
            "Vere dignum et justum est, æquum et salutáre, nos tibi semper et ubíque grátias ágere: Dómine sancte, Pater omnípotens, ætérne Deus: per Christum, Dóminum nostrum. Per quem majestátem tuam laudant Angeli, adórant Dominatiónes, tremunt Potestátes. Cæli cælorúmque Virtútes ac beáta Séraphim sócia exsultatióne concélebrant. Cum quibus et nostras voces ut admítti jubeas, deprecámur, súpplici confessione dicéntes:",
            "",
          ],
          "vernacular": [
            "Comum",
            "É verdadeiramente digno e justo, racional e salutar render-Vos graças em todos os lugares e sempre, ó Senhor santo, Pai omnipotente, eterno Deus, por nosso Senhor Jesus Cristo. Pelo qual os Anjos louvam a vossa majestade, as Dominações a adoram e as Potestades se prostram reverentes; e os Céus, as Virtudes dos céus e os bem-aventurados Serafins a celebram em comum alegria e louvor. Dignai-Vos permitir, Vos imploramos, que as nossas vozes suplicantes se unam às deles, dizendo:",
            "",
          ],
        },
        "id": "Prefatio",
      },
      {
        "body": {
          "latin": [
            "Sap 3:4-6",
            "Et si coram homínibus torménta passi sunt, Deus tentávit eos: tamquam aurum in fornáce probávit eos, et quasi holocáusta accépit eos.",
            "",
          ],
          "vernacular": [
            "Sb 3:4-6",
            "Se sofreram tormentos diante dos homens, foi para Deus os provar. Deus provou-os na fornalha, como ao ouro, e recebeu-os, como hóstia de holocausto.",
            "",
            "",
          ],
        },
        "id": "Communio",
      },
      {
        "body": {
          "latin": [
            "Quǽsumus, omnípotens Deus: ut, qui cœléstia aliménta percépimus, intercedéntibus beátis Martýribus tuis Vincéntio et Anastásio, per hæc contra ómnia advérsa muniámur.",
            "Per Dominum…",
            "",
          ],
          "vernacular": [
            "Ó Deus omnipotente, Vos suplicamos, fazei que, havendo nós recebido os alimentos celestiais, sejamos fortalecidos contra todas as adversidades por intercessão dos vossos B. B. Mártires Vicente e Anastácio.",
            "Por nosso Senhor...",
            "",
          ],
        },
        "id": "Postcommunio",
      },
    ],
  },
]
`;

exports[`2024-02-25 1`] = `
[
  {
    "info": {
      "colors": [
        "v",
      ],
      "date": "2024-02-25",
      "day": Day {
        "celebration": [
          Observance {
            "colors": [
              "v",
            ],
            "date": "2024-02-25",
            "flexibility": "tempora",
            "id": "tempora:Quad2-0:1:v",
            "name": "Quad2-0",
            "priority": 19,
            "rank": 1,
            "title": "2º Domingo da Quaresma",
            "weekday": 0,
          },
        ],
        "commemoration": [],
        "date": "2024-02-25",
        "tempora": [
          Observance {
            "colors": [
              "v",
            ],
            "date": "2024-02-25",
            "flexibility": "tempora",
            "id": "tempora:Quad2-0:1:v",
            "name": "Quad2-0",
            "priority": 19,
            "rank": 1,
            "title": "2º Domingo da Quaresma",
            "weekday": 0,
          },
        ],
      },
      "id": "tempora:Quad2-0:1:v",
      "rank": 1,
      "tempora": null,
      "title": "2º Domingo da Quaresma",
    },
    "sections": [
      {
        "body": {
          "latin": [
            "Ps 24:6; 24:3; 24:22",
            "Reminíscere miseratiónum tuarum, Dómine, et misericórdiæ tuæ, quæ a sǽculo sunt: ne umquam dominéntur nobis inimíci nostri: líbera nos, Deus Israël, ex ómnibus angústiis nostris.",
            "Ps 24:1-2",
            "Ad te, Dómine, levávi ánimam meam: Deus meus, in te confído, non erubéscam.",
            "Glória Patri…",
            "Reminíscere miseratiónum tuarum, Dómine, et misericórdiæ tuæ, quæ a sǽculo sunt: ne umquam dominéntur nobis inimíci nostri: líbera nos, Deus Israël, ex ómnibus angústiis nostris.",
            "",
          ],
          "subsections": [],
          "vernacular": [
            "Sl 24:6, 3, 22.",
            "Lembrai-Vos, Senhor, de que a vossa bondade e misericórdia são eternas! Que os nossos inimigos nunca triunfem de nós. Ó Deus de Israel, livrai-nos de todas nossas angústias.",
            "Sl 24:1-2",
            "A Vós, Senhor, elevei a minha alma: meu Deus, confio em Vós; não ficarei confundido.",
            "Glória ao Pai.",
            "Lembrai-Vos, Senhor, de que a vossa bondade e misericórdia são eternas! Que os nossos inimigos nunca triunfem de nós. Ó Deus de Israel, livrai-nos de todas nossas angústias.",
            "",
          ],
        },
        "id": "Introitus",
      },
      {
        "body": {
          "latin": [
            "Deus, qui cónspicis omni nos virtúte destítui: intérius exteriúsque custódi; ut ab ómnibus adversitátibus muniámur in córpore, et a pravis cogitatiónibus mundémur in mente.",
            "Per Dominum…",
            "",
          ],
          "subsections": [],
          "vernacular": [
            "Ó Deus, que conheceis como somos destituídos de toda a virtude, guardai-nos interior e exteriormente, a fim de que o nosso corpo seja preservado de todas as adversidades e a nossa alma purificada de todos os maus pensamentos.",
            "Por nosso Senhor...",
            "",
          ],
        },
        "id": "Oratio",
      },
      {
        "body": {
          "latin": [
            "Léctio Epístolæ beáti Pauli Apóstoli ad Thessalonicénses",
            "1 Thess 4:1-7.",
            "Fratres: Rogámus vos et obsecrámus in Dómino Jesu: ut, quemádmodum accepístis a nobis, quómodo opórteat vos ambuláre et placére Deo, sic et ambulétis, ut abundétis magis. Scitis enim, quæ præcépta déderim vobis Per Dominum Jesum. Hæc est enim volúntas Dei, sanctificátio vestra: ut abstineátis vos a fornicatióne, ut sciat unusquísque vestrum vas suum possidére in sanctificatióne et honóre; non in passióne desidérii, sicut et gentes, quæ ignórant Deum: et ne quis supergrediátur neque circumvéniat in negótio fratrem suum: quóniam vindex est Dóminus de his ómnibus, sicut prædíximus vobis et testificáti sumus. Non enim vocávit nos Deus in immundítiam, sed in sanctificatiónem: in Christo Jesu, Dómino nostro.",
            "",
          ],
          "subsections": [],
          "vernacular": [
            "Lição da Ep.ª do B. Ap.º Paulo aos Tessalonicenses.",
            "1 Ts 4:1-7",
            "Meus irmãos: Vos pedimos e exortamos, em nome do Senhor Jesus, que, havendo aprendido de nós como deveis conduzir-vos para agradar a Deus, tenhais uma conduta de modo a aperfeiçoar-vos cada vez mais. Com efeito, conheceis os preceitos que vos dei da parte do Senhor Jesus. O que Ele quer é a vossa santificação: que vos abstenhais da impureza carnal, para que cada um saiba guardar o vaso do seu corpo em santidade e honestidade, e não segundo os apetites das paixões, como os pagãos, que desconhecem Deus; e que ninguém engane o seu irmão, nem o prejudique, pois, o Senhor é vingador destas coisas, como já provámos e testemunhámos. Porquanto Deus vos não chamou para os gozos da carne, mas para a santificação em nosso Senhor Jesus Cristo.",
            "",
          ],
        },
        "id": "Lectio",
      },
      {
        "body": {
          "latin": [
            "Ps 24:17-18",
            "Tribulatiónes cordis mei dilatátæ sunt: de necessitátibus meis éripe me, Dómine,",
            "℣. Vide humilitátem meam et labórem meum: et dimítte ómnia peccáta mea.",
            "",
            "Tractus",
            "Ps 105:1-4",
            "Confitémini Dómino, quóniam bonus: quóniam in sǽculum misericórdia ejus.",
            "℣. Quis loquétur poténtias Dómini: audítas fáciet omnes laudes ejus?",
            "℣. Beáti, qui custódiunt judícium et fáciunt justítiam in omni témpore.",
            "℣. Meménto nostri, Dómine, in beneplácito pópuli tui: vísita nos in salutári tuo.",
            "",
          ],
          "subsections": [],
          "vernacular": [
            "Sl 24:17-18",
            "As tribulações do meu coração cresceram. Ó Senhor, livrai-me das minhas misérias.",
            "℣. Vede a minha humilhação e fadiga e perdoai os meus pecados.",
            "",
            "Tractus",
            "Sl 105:1-4",
            "Louvai o Senhor, pois Ele é bom: a sua misericórdia é eterna.",
            "℣. Quem será capaz de narrar as maravilhas da omnipotência do Senhor e apregoar os seus louvores?",
            "℣. Bem-aventurados aqueles que procedem com equidade e justiça em todas as ocasiões.",
            "℣. Pela vossa bondade para com vosso povo, Senhor, lembrai-Vos de nós: visitai-nos para alcançarmos a salvação.",
            "",
          ],
        },
        "id": "Graduale",
      },
      {
        "body": {
          "latin": [
            "Sequéntia ☩ sancti Evangélii secúndum Matthǽum",
            "Matt 17:1-9",
            "In illo témpore: Assúmpsit Jesus Petrum, et Jacóbum, et Joánnem fratrem eius, et duxit illos in montem excélsum seórsum: et transfigurátus est ante eos. Et resplénduit fácies ejus sicut sol: vestiménta autem ejus facta sunt alba sicut nix. Et ecce, apparuérunt illis Móyses et Elías cum eo loquéntes. Respóndens autem Petrus, dixit ad Jesum: Dómine, bonum est nos hic esse: si vis, faciámus hic tria tabernácula, tibi unum, Móysi unum et Elíæ unum. Adhuc eo loquénte, ecce, nubes lúcida obumbrávit eos. Et ecce vox de nube, dicens: Hic est Fílius meus diléctus, in quo mihi bene complácui: ipsum audíte. Et audiéntes discípuli, cecidérunt in fáciem suam, et timuérunt valde. Et accéssit Jesus, et tétigit eos, dixítque eis: Súrgite, et nolíte timére. Levántes autem óculos suos, néminem vidérunt nisi solum Jesum. Et descendéntibus illis de monte, præcépit eis Jesus, dicens: Némini dixéritis visiónem, donec Fílius hóminis a mórtuis resúrgat.",
            "",
          ],
          "subsections": [],
          "vernacular": [
            "Continuação ☩ do santo Evangelho segundo S. Mateus.",
            "Mt. 17:1-9",
            "Naquele tempo, Jesus levou consigo Pedro, Tiago e João e conduziu-os a um monte alto e separado, transfigurando-se ante eles: seu rosto resplandecia, como o sol, e os seus vestidos tornaram-se brancos, como a neve! E Moisés e Elias apareceram, conversando com Jesus. Então, Pedro disse a Jesus: «Senhor, é tão bom estar aqui!... Se quereis, façamos aqui três tendas: uma para Vós, outra para Moisés e outra para Elias!». Ainda ele falava, eis que uma nuvem brilhante os envolveu, saindo do seio dela uma voz, que dizia: «Este é o meu Filho muito amado, em quem pus as minhas complacências; ouvi-O». Havendo escutado a voz, os discípulos caíram com o rosto no chão e ficaram atemorizados. Mas Jesus tocou-os e disse-lhes: «Levantai-vos; não vos amedronteis». Então, erguendo os olhos, já nada viram senão só Jesus. Desceram do monte, dando-lhes Jesus esta ordem: «Não conteis a ninguém esta visão até que o Filho do homem ressuscite dos mortos».",
            "",
          ],
        },
        "id": "Evangelium",
      },
      {
        "body": {
          "latin": [
            "Ps 118:47; 118:48",
            "Meditábor in mandátis tuis, quæ diléxi valde: et levábo manus meas ad mandáta tua, quæ diléxi.",
            "",
          ],
          "subsections": [],
          "vernacular": [
            "Sl 118:47-48",
            "Meditarei nos vossos Mandamentos, que muito amo: e levantarei as minhas mãos para cumprir esses Mandamentos, que, repito, muito amo.",
            "",
          ],
        },
        "id": "Offertorium",
      },
      {
        "body": {
          "latin": [
            "Sacrifíciis præséntibus, Dómine, quǽsumus, inténde placátus: ut et devotióni nostræ profíciant et salúti.",
            "Per Dominum…",
            "",
          ],
          "subsections": [],
          "vernacular": [
            "Senhor, Vos suplicamos, dignai-Vos olhar benigno para o presente sacrifício, a fim de que sirva de proveito à nossa piedade e à nossa salvação.",
            "Por nosso Senhor...",
            "",
          ],
        },
        "id": "Secreta",
      },
      {
        "body": {
          "latin": [
            "de Quadragesima",
            "Vere dignum et justum est, æquum et salutáre, nos tibi semper et ubíque grátias ágere: Dómine, sancte Pater, omnípotens ætérne Deus: Qui corporáli ieiúnio vítia cómprimis, mentem élevas, virtútem largíris et prǽmia: per Christum Dóminum nostrum. Per quem majestátem tuam laudant Angeli, adórant Dominatiónes, tremunt Potestátes. Cæli cælorúmque Virtútes, ac beáta Séraphim, sócia exsultatióne concélebrant. Cum quibus et nostras voces, ut admítti iúbeas, deprecámur, súpplici confessióne dicéntes:",
            "",
          ],
          "subsections": [],
          "vernacular": [
            "da Quaresma",
            "Verdadeiramente é digno e justo, racional e salutar dar-Vos graças em todos os lugares e sempre, Senhor santo, Pai omnipotente, eterno Deus, pois Vós, pelo jejum natural, reprimis os vícios, elevais o espírito e concedeis-nos a virtude e o prémio por Cristo, nosso Senhor: pelo qual os Anjos louvam a vossa majestade, as Dominações a adoram e as Potestades trémulas a reverenciam; e os Céus, as Virtudes dos Céus e os bem-aventurados Serafins se associam em comum louvor. Dignai-Vos permitir, Senhor, que as nossas vozes suplicantes se unam às deles, dizendo:",
            "",
          ],
        },
        "id": "Prefatio",
      },
      {
        "body": {
          "latin": [
            "Ps 5:2-4",
            "Intéllege clamórem meum: inténde voci oratiónis meæ, Rex meus et Deus meus: quóniam ad te orábo, Dómine.",
            "",
          ],
          "subsections": [],
          "vernacular": [
            "Sl 5:2-4",
            "Ouvi o meu clamor; ouvi a minha oração, ó meu Rei, ó meu Deus; pois orei a Vós.",
            "",
          ],
        },
        "id": "Communio",
      },
      {
        "body": {
          "latin": [
            "Súpplices te rogámus, omnípotens Deus: ut quos tuis réficis sacraméntis, tibi étiam plácitis móribus dignánter deservíre concédas.",
            "Per Dominum…",
            "",
          ],
          "subsections": [],
          "vernacular": [
            "Ó Deus omnipotente, Vos suplicamos instantemente que aqueles a quem sustentais com vossos sacramentos tenham uma conduta de vida que Vos seja agradável.",
            "Por nosso Senhor...",
            "",
          ],
        },
        "id": "Postcommunio",
      },
    ],
  },
]
`;
